### ISSUE TITLE
loadDatabaseRules unable to return database emulator error message

### ISSUE THREAD

**Issue by mikehardy** [firebase/firebase-js-sdk#4914]
### [REQUIRED] Describe your environment

  * Operating System version: macOS 11.3.1
  * Browser version: testing utilties, node 14
  * Firebase SDK version: `@firebase/testing` v1.3.1 against emulator in firebase-tools 9.10.1
  * Firebase Product: database emulator interacting with testing (auth, database, storage, etc)

#### Steps to reproduce:

Send incorrect rules in to the database emulator via `loadDatabaseRules({ databaseName: 'foo', rules: "{rules: 'singleQuotesAreBad': "{".write": true,".read":false}})`

Note that single quotes aren't correct for JSON, the database emulator is unhappy about this but examine the response, note the extra space, visible by hacking in a console output:

```
detox[33607] ERROR: at ../node_modules/@firebase/rules-unit-testing/dist/index.cjs.js:414:33 
 got an error? {
  "error" : "1:100: Key names can't contain \".\", \"#\", \"$\", \"/\", \"[\", or \"]\" (unbound names start with \"$\")\n"
```

Do you see that space after `"error"` and the `:`?

So you get this instead of an error bubbling up:

```
      at JSON.parse (<anonymous>)
      at Object.<anonymous> (/Users/mike/work/Invertase/react-native-firebase/node_modules/@firebase/rules-unit-testing/dist/index.cjs.js:415:46)
```

#### Relevant Code:

Code trace demonstrated above

This is probably a database emulator more than a testing error but (they may be interested as well in firebase-tools repo?) but the code here should tolerate non-JSON error messages and send the raw string as well I think?

**Comment by looptheloop88** [comment]
Hi @mikehardy, I can't reproduce the similar error that you're getting. The exact code snippet that you've shared will cause a `SyntaxError: Unexpected token '{'` error.

On the other hand, if I use this [implementation](https://github.com/firebase/quickstart-testing/blob/master/unit-test-security-rules/test/database.spec.js#L51-L55) and just update the database security rules with key enclosed by single quotes, I will get a `SyntaxError: Unexpected token u in JSON at position 0`  error.

Rules:
```
{
  "rules": {
    'singleQuotesAreBad': {
      ".write": true,
      ".read": false
    }
  }
}
```

If I can replicate the issue, I can have a better look into it. Would you mind sharing a [minimal, but complete](https://stackoverflow.com/help/minimal-reproducible-example) sample of a project that I can run locally to replicate the issue?

**Comment by mikehardy** [comment]
Thanks for looking @looptheloop88 
I was using 9.10.1 of firebase-tools and had an unrelated issue that was fixed with 9.10.2, it may be that they are returning the error message without the spurious space in `error :` so it parses now. Appeared to me at first glance to be one of those in-between API bugs where this package could handle malformed input better, but firebase-tools should not have been emitting bad output.
Still trying to finish a few things up before GoogleIO so this might sit for a little while, apologies

**Comment by looptheloop88** [comment]
I just checked that I am using `firebase-tools` version 9.10.2, the latest one. This might be the reason why I was not able to replicate the issue.

I will close this issue now, but you can reopen it if you encounter the error again in the future.
